src/aspire/basis/fb.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import numpy as np
+
+from aspire.basis.basis_utils import num_besselj_zeros
+from aspire.utils.matlab_compat import m_reshape
+
+logger = logging.getLogger(__name__)
+
+
+class FBBasisMixin(object):
+    """
+    FBBasisMixin is a mixin implementing methods specific to Fourier-Bessel expansions,
+    to be inherited by Fourier-Bessel subclasses of Basis.
+    """
+
+    def _getfbzeros(self):
+        """
+        Generate zeros of Bessel functions
+        """
+        # get upper_bound of zeros of Bessel functions
+        upper_bound = min(self.ell_max + 1, 2 * self.nres + 1)
+
+        # List of number of zeros
+        n = []
+        # List of zero values (each entry is an ndarray; all of possibly different lengths)
+        zeros = []
+
+        # generate zeros of Bessel functions for each ell
+        for ell in range(upper_bound):
+            # for each ell, num_besselj_zeros returns the zeros of the
+            # order ell Bessel function which are less than 2*pi*c*R = nres*pi/2,
+            # the truncation rule for the Fourier-Bessel expansion
+            _n, _zeros = num_besselj_zeros(
+                ell + (self.ndim - 2) / 2, self.nres * np.pi / 2
+            )
+            if _n == 0:
+                break
+            else:
+                n.append(_n)
+                zeros.append(_zeros)
+
+        #  get maximum number of ell
+        self.ell_max = len(n) - 1
+
+        #  set the maximum of k for each ell
+        self.k_max = np.array(n, dtype=int)
+
+        max_num_zeros = max(len(z) for z in zeros)
+        for i, z in enumerate(zeros):
+            zeros[i] = np.hstack(
+                (z, np.zeros(max_num_zeros - len(z), dtype=self.dtype))
+            )
+
+        self.r0 = m_reshape(np.hstack(zeros), (-1, self.ell_max + 1)).astype(self.dtype)
```

Comment:
This is just a Matlab artifact. Let's make an issue to change this to a list of arrays. No need for this zero-padding.

Reply:
Just to be clear, for the code below:

https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/6cb389edf21e811224cc98f4fde0d63ed778b9a7/src/aspire/basis/fb_2d.py#L140-L145

It would change the `[k-1, ell]` to `[k-1][ell]` since we would be getting the index from the list, and then index from the 1d `np.array`?

Reply:
created #609